omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,89 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/identifiersource", supportedClass = IdentifierSource.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" , "2.1.*" })
+public class IdentifierSourceResource extends DelegatingCrudResource<IdentifierSource>{
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		
+		DelegatingResourceDescription description = null;
+		
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addRequiredProperty("name");
+			description.addRequiredProperty("description");
+			description.addRequiredProperty("identifierType");
+		}
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		
+		//IdentifierSource is an abstract class therefore we cannot instantiate it. Please comment on the approach below.
+		
+		IdentifierSource obj = null;
+		
+		return (IdentifierSource) obj;
+	}
+	
+	@Override
+	public IdentifierSource save(IdentifierSource delegate) {
+		return Context.getService(IdentifierSourceService.class).saveIdentifierSource(delegate);
+	}
+	
+	@Override
+	public IdentifierSource getByUniqueId(String uniqueId) {
+		return Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uniqueId);
+	}
+	
+	@Override
+    protected void delete(IdentifierSource delegate, String reason, RequestContext context) throws ResponseException {
+		Context.getService(IdentifierSourceService.class).purgeIdentifierSource(delegate);
+    }
+	
+	@Override
+	public void purge(IdentifierSource delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException("purge not supported");
```

Comment:
Are you sure we do not supporting purging?

Reply:
I am using the `purgeIdentifierSource(IdentifierSource)` method in the `delete()` resource.

Reply:
That one belongs to purge

Reply:
fixed